# HG changeset patch
# User Jorg K <jorgk@jorgk.com>
# Date 1629193378 -7200
# Parent  bf11f0f6a6929ba968f472fb56d74e039ada484d
Branding 8: Sign setup.exe and all .exe and .dll files.

diff --git a/browser/installer/windows/sign.py b/browser/installer/windows/sign.py
new file mode 100644
--- /dev/null
+++ b/browser/installer/windows/sign.py
@@ -0,0 +1,20 @@
+import subprocess, sys
+
+p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
+    "Set-AuthenticodeSignature -FilePath instgen\\setup.exe -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert)"],
+    stdout=sys.stdout)
+p.communicate()
+
+p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
+    "Get-ChildItem -Path ..\\..\\..\\..\\dist\\betterbird -Filter *.exe -File | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\..\\dist\\betterbird\\$_ -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert) } "],
+    stdout=sys.stdout)
+p.communicate()
+p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
+    "Get-ChildItem -Path ..\\..\\..\\..\\dist\\bin -Filter betterbird.exe -File | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\..\\dist\\bin\\$_ -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert) } "],
+    stdout=sys.stdout)
+p.communicate()
+
+p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
+    "Get-ChildItem -Path ..\\..\\..\\..\\dist\\betterbird -Filter *.dll -File | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\..\\dist\\betterbird\\$_ -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert) } "],
+    stdout=sys.stdout)
+p.communicate()
diff --git a/toolkit/mozapps/installer/windows/nsis/makensis.mk b/toolkit/mozapps/installer/windows/nsis/makensis.mk
--- a/toolkit/mozapps/installer/windows/nsis/makensis.mk
+++ b/toolkit/mozapps/installer/windows/nsis/makensis.mk
@@ -57,16 +57,18 @@ CUSTOM_UI = \
 	$(INSTALL) $(addprefix $(MOZILLA_DIR)/other-licenses/nsis/,$(CUSTOM_UI)) $(CONFIG_DIR)
 	cd $(CONFIG_DIR) && $(call WINEWRAP,$(MAKENSISU)) $(MAKENSISU_FLAGS) installer.nsi
 ifdef MOZ_STUB_INSTALLER
 	cd $(CONFIG_DIR) && $(call WINEWRAP,$(MAKENSISU)) $(MAKENSISU_FLAGS) stub.nsi
 endif
 
 ifdef ZIP_IN
 installer:: $(CONFIG_DIR)/setup.exe $(ZIP_IN)
+	@echo 'Signing.'
+	$(PYTHON3) $(topsrcdir)/browser/installer/windows/sign.py
 	@echo 'Packaging $(WIN32_INSTALLER_OUT).'
 	$(NSINSTALL) -D '$(ABS_DIST)/$(PKG_INST_PATH)'
 	$(PYTHON3) $(MOZILLA_DIR)/mach repackage installer \
 	  -o '$(ABS_DIST)/$(PKG_INST_PATH)$(PKG_INST_BASENAME).exe' \
 	  --package-name '$(MOZ_PKG_DIR)' \
 	  --package '$(ZIP_IN)' \
 	  --tag $(topsrcdir)/$(MOZ_BUILD_APP)/installer/windows/app.tag \
 	  --setupexe $(CONFIG_DIR)/setup.exe \
